### ISSUE TITLE
flyc_attn_fwd on gfx950 returns wrong results (bf16) and NaN (fp16) for large logits (0.14.1b)

### ISSUE THREAD

**Issue by jeffdaily** [ROCm/aotriton#246]
PyTorch's reland of AOTriton 0.14.1b (pytorch/pytorch#197747) was auto-reverted after `test_mem_eff_attention_single_query_tail` failed on gfx950 (MI350). I reproduced it on gfx950 (HIP 7.15) by swapping only the AOTriton library under one PyTorch build: 0.13b (Triton `attn_fwd`) passes, and 0.14.1b fails. Kernel traces with `AMD_LOG_LEVEL=3` show 0.14.1b dispatching these shapes to `flyc_attn_fwd`.

What triggers it is the softmax scale. With the default `1/sqrt(d)`, the error is close to 0.13b. With `scale=1.0` (logits ~ sqrt(d)):

- bf16: max error 0.06 to 0.24 vs 0.010 to 0.015 on 0.13b (fp32 reference), for head_dim 64 to 512, causal and non-causal, across all query rows and heads. bf16 head_dim 128 dispatches to `aiter::fmha_fwd_hd128_bf16` instead and is correct.
- fp16: NaN outputs for head_dim 64, 128, 256 (5127 NaNs at h=40, q=kv=289, d=128); head_dim 512 has no NaN but 15x the 0.13b error.

Repro:

```python
import torch, torch.nn.functional as F
from torch.nn.attention import sdpa_kernel, SDPBackend

torch.manual_seed(0)
for dtype in (torch.bfloat16, torch.float16):
    q, k, v = (torch.randn(1, 40, 289, 128, device="cuda", dtype=dtype) for _ in range(3))
    with sdpa_kernel(backends=[SDPBackend.MATH]):
        ref = F.scaled_dot_product_attention(q.float(), k.float(), v.float(), scale=1.0)
    with sdpa_kernel(backends=[SDPBackend.EFFICIENT_ATTENTION]):
        out = F.scaled_dot_product_attention(q, k, v, scale=1.0)
    print(dtype, (out.float() - ref).abs().max().item(), out.isnan().sum().item())
```

On gfx950 with 0.14.1b this prints NaN for fp16. Use head_dim 256 to see the bf16 error, since bf16 at head_dim 128 routes to aiter. The release notes say flyc also ships for gfx1201; I have not tested that arch.
